# Generate .csv tables of Calibration steps

In [15]:
#make sure s = 1 in './data/datamapping.py' file 
!python3 main.py

* * datamapping module loaded!
* * a_utils module loaded!
* * b_data_transform module loaded!
* * c_pl_pw module loaded!
* * d_del_del module loaded!
* * e_error_estimation module loaded!
* * f_star_wise module loaded!
* * g_result module loaded!
* * main_modules module loaded!
* * datamapping module loaded!
* * * * * * * * * * * * * * * * * * * * 
* * * * * * * * * * * * * * * * * * * * 


Master Thesis Project: 	Galactic BVIJHK Leavitt Law Calibration for R_v = 3.23 

  			To Refine systematic errors in luminosity, distance and reddening of individual Cepheid. 
			______________________________________________________

     			Author: Shubham Mamgain (mshubham00@gmail.com) 
     			Supervisor I: Dr. Jesper Storm (AIP Potsdam)
     			Supervisor II: Prof. Dr. Maria Rosa Cioni




			To begin the calibration process, store cleaned data at ./data/input/<file_name>.csv 

			For datafile selection (k) and columns mapping, edit ./data/datamapping.py file.

			Autosave (s = 1) | Autoprocess

In [2]:
import pandas as pd

files = {}
files['0'] = '59_madore_3.23'
files['1'] = '76_IRSB_IJ_HK_3.23'#'76_IRSB_IJ_HK_3.23'#'94_jesper_3.23'
files['2'] = '20_cluster_cruz_3.23'
files['3'] = '29_LMC_3.4'
files['4'] = '32_SMC_VIJK_2.53'
files['5'] = '103_raw_data_IRSB_3.23'
files['6'] = '109_IRSB_IH_VJ'

reg={}
#for k in range(6):
#    reg[str(k)] = pd.read_csv(f'data/processed/{files[str(k)]}/2_PLPW/{files[str(k)][:2]}_5_regression.csv')

In [3]:
from data.datamapping import R, R123, dis_flag, mag, colors, A
print(R, '\n', A)
for d,dis in enumerate(dis_flag):
    for a,c1 in enumerate(mag):
        for b,c2 in enumerate(mag[a+1:]):
            x = 1
            print([f'{x+c1+c2}: {R123(x,c1,c2) :.3f}' for x in mag])

* * datamapping module loaded!
{'B': 4.061402, 'V': 3.23, 'I': 1.9670699999999999, 'J': 0.9583410000000001, 'H': 0.586568, 'K': 0.397613} 
 {'B': 1.2574, 'V': 1.0, 'R': 0.845, 'I': 0.609, 'J': 0.2967, 'H': 0.1816, 'K': 0.1231}
['BBV: 4.887', 'VBV: 3.887', 'IBV: 2.367', 'JBV: 1.153', 'HBV: 0.706', 'KBV: 0.478']
['BBI: 1.940', 'VBI: 1.543', 'IBI: 0.939', 'JBI: 0.458', 'HBI: 0.280', 'KBI: 0.190']
['BBJ: 1.309', 'VBJ: 1.041', 'IBJ: 0.634', 'JBJ: 0.309', 'HBJ: 0.189', 'KBJ: 0.128']
['BBH: 1.169', 'VBH: 0.930', 'IBH: 0.566', 'JBH: 0.276', 'HBH: 0.169', 'KBH: 0.114']
['BBK: 1.109', 'VBK: 0.882', 'IBK: 0.537', 'JBK: 0.262', 'HBK: 0.160', 'KBK: 0.109']
['BVI: 3.216', 'VVI: 2.557', 'IVI: 1.557', 'JVI: 0.759', 'HVI: 0.464', 'KVI: 0.315']
['BVJ: 1.788', 'VVJ: 1.422', 'IVJ: 0.866', 'JVJ: 0.422', 'HVJ: 0.258', 'KVJ: 0.175']
['BVH: 1.537', 'VVH: 1.222', 'IVH: 0.744', 'JVH: 0.363', 'HVH: 0.222', 'KVH: 0.150']
['BVK: 1.434', 'VVK: 1.141', 'IVK: 0.695', 'JVK: 0.338', 'HVK: 0.207', 'KVK: 0.140']
['BIJ: 4

In [21]:
for a,c1 in enumerate(mag):
    for b,c2 in enumerate(mag[a+1:]):
        x=1
        print(f'{mag[0]+c1+c2} & {R123(mag[0],c1,c2)} & {mag[1]+c1+c2} & {R123(mag[1],c1,c2)} & {mag[2]+c1+c2} & {R123(mag[2],c1,c2)} & {mag[3]+c1+c2} & {R123(mag[3],c1,c2)} & {mag[4]+c1+c2} & {R123(mag[4],c1,c2)} & {mag[5]+c1+c2} & {R123(mag[5],c1,c2)} \\\\') 
        #print([f'{x+c1+c2}: {R123(x,c1,c2) :.3f}' for x in mag])

BBV & 4.887 & VBV & 3.887 & IBV & 2.367 & JBV & 1.153 & HBV & 0.706 & KBV & 0.478 \\
BBI & 1.94 & VBI & 1.543 & IBI & 0.939 & JBI & 0.458 & HBI & 0.28 & KBI & 0.19 \\
BBJ & 1.309 & VBJ & 1.041 & IBJ & 0.634 & JBJ & 0.309 & HBJ & 0.189 & KBJ & 0.128 \\
BBH & 1.169 & VBH & 0.93 & IBH & 0.566 & JBH & 0.276 & HBH & 0.169 & KBH & 0.114 \\
BBK & 1.108 & VBK & 0.882 & IBK & 0.537 & JBK & 0.262 & HBK & 0.16 & KBK & 0.108 \\
BVI & 3.216 & VVI & 2.557 & IVI & 1.557 & JVI & 0.759 & HVI & 0.464 & KVI & 0.315 \\
BVJ & 1.788 & VVJ & 1.422 & IVJ & 0.866 & JVJ & 0.422 & HVJ & 0.258 & KVJ & 0.175 \\
BVH & 1.537 & VVH & 1.222 & IVH & 0.744 & JVH & 0.363 & HVH & 0.222 & KVH & 0.15 \\
BVK & 1.434 & VVK & 1.14 & IVK & 0.694 & JVK & 0.338 & HVK & 0.207 & KVK & 0.14 \\
BIJ & 4.025 & VIJ & 3.201 & IIJ & 1.95 & JIJ & 0.95 & HIJ & 0.581 & KIJ & 0.394 \\
BIH & 2.943 & VIH & 2.341 & IIH & 1.425 & JIH & 0.694 & HIH & 0.425 & KIH & 0.288 \\
BIK & 2.587 & VIK & 2.057 & IIK & 1.253 & JIK & 0.61 & HIK & 0.374 & KIK & 

In [20]:
mag = ['B', 'V', 'I','J','H','K']
mag = ['V', 'I','J','K']
k=1
data = reg[str(k)]
for m in mag:
    sl = data[f'{m}0'].iloc[0]
    serr = data[f'{m}0'].iloc[2]
    intr = data[f'{m}0'].iloc[1]
    ierr = data[f'{m}0'].iloc[3]
    #print(f'{m} & {sl:.2f} ({serr:.2f}) & {intr:.2f} ({ierr:.2f})\\\\')
    print(f'{m} = ({sl:.3f} \pm {serr:.3f}) (\log P -1) + {intr:.3f} \pm {ierr:.3f} \\\\')



V = (-2.281 \pm 0.016) (\log P -1) + -3.911 \pm 0.004 \\
I = (-2.581 \pm 0.015) (\log P -1) + -4.691 \pm 0.004 \\
J = (-2.799 \pm 0.007) (\log P -1) + -5.177 \pm 0.002 \\
K = (-2.981 \pm 0.002) (\log P -1) + -5.613 \pm 0.001 \\


"\nslp = r[[f'{m}0' for m in mag]].iloc[0]\nprint(f'Slope & {slp[0]:.2f} ({esl[0]:.3f}) & {slp[1]:.2f} ({esl[1]:.3f}) & {slp[2]:.2f} ({esl[2]:.3f}) & {slp[3]:.2f} ({esl[3]:.3f}) & {slp[4]:.2f} ({esl[4]:.3f}) & {slp[5]:.2f} ({esl[5]:.3f}) \\\\')\nprint(f'Intercept & {inr[0]:.2f} ({ein[0]:.3f}) & {inr[1]:.2f} ({ein[1]:.3f}) & {inr[2]:.2f} ({ein[2]:.3f}) & {inr[3]:.2f} ({ein[3]:.3f}) & {inr[4]:.2f} ({ein[4]:.3f}) & {inr[5]:.2f} ({ein[5]:.3f}) \\\\')\n# wesenheit PW\n\ncols = colors(mag)\nfor col in cols:#['BK', 'BI', 'VI', 'JK']:\n    slp = r[[f'{m}{col}' for m in mag]].iloc[0]\n    inr = r[[f'{m}{col}' for m in mag]].iloc[1]\n    esl = r[[f'{m}{col}' for m in mag]].iloc[2]\n    ein = r[[f'{m}{col}' for m in mag]].iloc[3]\n    #print(f'{col} & {slp[0]:.2f} ({esl[0]:.3f}) & {slp[1]:.2f} ({esl[1]:.3f}) & {slp[2]:.2f} ({esl[2]:.3f}) & {slp[3]:.2f} ({esl[3]:.3f}) & {slp[4]:.2f} ({esl[4]:.3f}) & {slp[5]:.2f} ({esl[5]:.3f}) \\\\')\n    #print(f'{col} & {inr[0]:.2f} ({ein[0]:.3f}) & {inr[1]:.2f}

In [ ]:
'''
slp = r[[f'{m}0' for m in mag]].iloc[0]
print(f'Slope & {slp[0]:.2f} ({esl[0]:.3f}) & {slp[1]:.2f} ({esl[1]:.3f}) & {slp[2]:.2f} ({esl[2]:.3f}) & {slp[3]:.2f} ({esl[3]:.3f}) & {slp[4]:.2f} ({esl[4]:.3f}) & {slp[5]:.2f} ({esl[5]:.3f}) \\\\')
print(f'Intercept & {inr[0]:.2f} ({ein[0]:.3f}) & {inr[1]:.2f} ({ein[1]:.3f}) & {inr[2]:.2f} ({ein[2]:.3f}) & {inr[3]:.2f} ({ein[3]:.3f}) & {inr[4]:.2f} ({ein[4]:.3f}) & {inr[5]:.2f} ({ein[5]:.3f}) \\\\')
# wesenheit PW

cols = colors(mag)
for col in cols:#['BK', 'BI', 'VI', 'JK']:
    slp = r[[f'{m}{col}' for m in mag]].iloc[0]
    inr = r[[f'{m}{col}' for m in mag]].iloc[1]
    esl = r[[f'{m}{col}' for m in mag]].iloc[2]
    ein = r[[f'{m}{col}' for m in mag]].iloc[3]
    #print(f'{col} & {slp[0]:.2f} ({esl[0]:.3f}) & {slp[1]:.2f} ({esl[1]:.3f}) & {slp[2]:.2f} ({esl[2]:.3f}) & {slp[3]:.2f} ({esl[3]:.3f}) & {slp[4]:.2f} ({esl[4]:.3f}) & {slp[5]:.2f} ({esl[5]:.3f}) \\\\')
    #print(f'{col} & {inr[0]:.2f} ({ein[0]:.3f}) & {inr[1]:.2f} ({ein[1]:.3f}) & {inr[2]:.2f} ({ein[2]:.3f}) & {inr[3]:.2f} ({ein[3]:.3f}) & {inr[4]:.2f} ({ein[4]:.3f}) & {inr[5]:.2f} ({ein[5]:.3f}) \\\\')

for col in ['BK', 'VJ']:
    for m in ['B', 'V', 'I','J','H','K']:
        sl = r[1][f'{m}0{col}{f}'].iloc[0]
        serr = r[1][f'{m}0{col}{f}'].iloc[2]
        intr = r[1][f'{m}0{col}{f}'].iloc[1]
        ierr = r[1][f'{m}0{col}{f}'].iloc[3]
        print(f'M_{m} & = {sl:.2f}(\log P - 1)(\pm {serr:.3f})  {intr:.2f}(\pm {ierr:.3f})\\\\')
'''

In [28]:
from lvtlaw.h_loadoutput import *
from data.datamapping import *
result, r_reg, res, pre = calibrated_result_()
r=r_reg

def print_PL(r, file_name, col):
    slp = r[[f'{m}{col}' for m in mag]].iloc[0]
    inr = r[[f'{m}{col}' for m in mag]].iloc[1]
    esl = r[[f'{m}{col}' for m in mag]].iloc[2]
    ein = r[[f'{m}{col}' for m in mag]].iloc[3]
    stt = r[[f'{m}{col}' for m in mag]].iloc[4]
    print('Raw Leavitt Law', file_name, col)
    for i,m in enumerate(mag):
        #print(f'M_{m}\,({stt[i]:.3f}) &= ({slp[i]:.3f} \pm {esl[i]:.3f})(\log P -1) {inr[i]:.3f} \pm {ein[i]:.3f}\\\\')
    #print(f'M_{m} & = ({slp:.3f} \pm {esl:.3f})(\log P - 1)  {inr:.2f} \pm {ein:.3f} \\\\')
    #print(f"M_{{{m}}} &= ({slp:.3f} \\pm {esl:.3f})(\\log P - 1) {inr:.3f} \\pm {ein:.3f} \\\\")
        print(f'{slp[i]:.3f} | {stt[i]:.3f} | {inr[i]:.3f} ')

for i in wes_show:
    print_PL(r_reg, file_name, f'0{i}S')
    

Raw Leavitt Law 59_madore 0BVS
-2.270 | 0.054 | -3.214 
-2.650 | 0.059 | -3.944 
-2.847 | 0.142 | -4.394 
-2.944 | 0.267 | -4.704 
-3.144 | 0.464 | -5.254 
-3.274 | 0.528 | -5.554 
-3.314 | 0.564 | -5.655 
Raw Leavitt Law 59_madore 0BRS
-2.265 | 0.057 | -3.214 
-2.654 | 0.121 | -3.944 
-2.857 | 0.170 | -4.396 
-2.963 | 0.270 | -4.706 
-3.174 | 0.411 | -5.258 
-3.308 | 0.456 | -5.558 
-3.351 | 0.485 | -5.658 
Raw Leavitt Law 59_madore 0BIS
-2.270 | 0.085 | -3.214 
-2.664 | 0.083 | -3.944 
-2.869 | 0.105 | -4.395 
-2.979 | 0.135 | -4.706 
-3.196 | 0.209 | -5.257 
-3.332 | 0.228 | -5.558 
-3.376 | 0.242 | -5.658 
Raw Leavitt Law 59_madore 0BJS
-2.272 | 0.088 | -3.213 
-2.666 | 0.060 | -3.944 
-2.872 | 0.066 | -4.395 
-2.982 | 0.083 | -4.705 
-3.198 | 0.116 | -5.257 
-3.334 | 0.124 | -5.557 
-3.378 | 0.131 | -5.657 
Raw Leavitt Law 59_madore 0BHS
-2.269 | 0.094 | -3.214 
-2.663 | 0.075 | -3.944 
-2.868 | 0.088 | -4.395 
-2.978 | 0.115 | -4.706 
-3.194 | 0.155 | -5.257 
-3.330 | 0.154 | -5.

In [23]:
r_reg.style
#!python3 main.py

In [ ]:
M_B\,(0.104) &= (-3.342 \pm 0.038)(\log P -1) -5.393 \pm 0.010\\
M_V\,(0.050) &= (-3.588 \pm 0.018)(\log P -1) -5.737 \pm 0.005\\
M_I\,(0.038) &= (-3.598 \pm 0.014)(\log P -1) -5.815 \pm 0.004\\
M_J\,(0.050) &= (-3.588 \pm 0.018)(\log P -1) -5.737 \pm 0.005\\
M_H\,(0.045) &= (-3.649 \pm 0.016)(\log P -1) -5.905 \pm 0.004\\
M_K\,(0.054) &= (-3.656 \pm 0.020)(\log P -1) -5.860 \pm 0.005\\

In [20]:
import pandas as pd
from data.datamapping import mag, col_lin, colors
files = {}
files['0'] = '59_madore_3.23'
files['1'] = '94_jesper_3.23'
files['2'] = '20_cluster_cruz_3.23'
files['3'] = '30_LMC_3.41'
files['4'] = '32_SMC_VIJK_2.74'
files['5'] = '103_raw_data_IRSB_3.23'
k=1
reg = pd.read_csv(f'data/processed/{files[str(k)]}/3_deldel/{files[str(k)][:2]}_del_slope_intercept.csv')
cols = colors(mag)
for col in cols:
    print(f"\t {col} & {reg[f'B0{col[0]}{col}'].iloc[0] : .3f} ({reg[f'B0{col[0]}{col}'].iloc[2] : .3f}) & {reg[f'V0{col[0]}{col}'].iloc[0] : .3f} ({reg[f'V0{col[0]}{col}'].iloc[2] : .3f}) & {reg[f'I0{col[0]}{col}'].iloc[0] : .3f} ({reg[f'I0{col[0]}{col}'].iloc[2] : .3f}) & {reg[f'J0{col[0]}{col}'].iloc[0] : .3f} ({reg[f'J0{col[0]}{col}'].iloc[2] : .3f}) & {reg[f'H0{col[0]}{col}'].iloc[0] : .3f} ({reg[f'H0{col[0]}{col}'].iloc[2] : .3f}) & {reg[f'K0{col[0]}{col}'].iloc[0] : .3f} ({reg[f'K0{col[0]}{col}'].iloc[2] : .3f}) \\\\")
    print('\t \\rowcolor{yellow!20}')
    print(f"\t    & {reg[f'B0B{col}'].iloc[0] : .3f} ({reg[f'B0B{col}'].iloc[2] : .3f}) & {reg[f'V0V{col}'].iloc[0] : .3f} ({reg[f'V0V{col}'].iloc[2] : .3f}) & {reg[f'I0I{col}'].iloc[0] : .3f} ({reg[f'I0I{col}'].iloc[2] : .3f}) & {reg[f'J0J{col}'].iloc[0] : .3f} ({reg[f'J0J{col}'].iloc[2] : .3f}) & {reg[f'H0H{col}'].iloc[0] : .3f} ({reg[f'H0H{col}'].iloc[2] : .3f}) & {reg[f'K0K{col}'].iloc[0] : .3f} ({reg[f'K0K{col}'].iloc[2] : .3f}) \\\\")
    print('\t \\hline')
    #print(f"\t{col} & {reg[f'B0B{col}'].iloc[0] : .3f} ({reg[f'B0B{col}'].iloc[2] : .3f})")

	 BV &  0.243 ( 0.084) &  0.398 ( 0.067) &  0.471 ( 0.056) &  0.479 ( 0.050) &  0.478 ( 0.045) &  0.471 ( 0.046) \\
	 \rowcolor{yellow!20}
	    &  0.243 ( 0.084) &  0.398 ( 0.067) &  0.653 ( 0.054) &  0.870 ( 0.035) &  0.929 ( 0.024) &  0.962 ( 0.017) \\
	 \hline
	 BI &  0.652 ( 0.092) &  0.758 ( 0.066) &  0.831 ( 0.045) &  0.798 ( 0.040) &  0.773 ( 0.035) &  0.765 ( 0.036) \\
	 \rowcolor{yellow!20}
	    &  0.652 ( 0.092) &  0.683 ( 0.068) &  0.832 ( 0.045) &  0.944 ( 0.025) &  0.968 ( 0.016) &  0.982 ( 0.011) \\
	 \hline
	 BJ &  0.881 ( 0.090) &  0.945 ( 0.060) &  0.975 ( 0.041) &  0.972 ( 0.021) &  0.938 ( 0.014) &  0.933 ( 0.016) \\
	 \rowcolor{yellow!20}
	    &  0.881 ( 0.090) &  0.835 ( 0.066) &  0.920 ( 0.041) &  0.972 ( 0.021) &  0.981 ( 0.014) &  0.990 ( 0.009) \\
	 \hline
	 BH &  0.917 ( 0.096) &  0.983 ( 0.065) &  1.015 ( 0.045) &  1.017 ( 0.025) &  0.988 ( 0.014) &  0.984 ( 0.014) \\
	 \rowcolor{yellow!20}
	    &  0.917 ( 0.096) &  0.859 ( 0.070) &  0.941 ( 0.043) &  0.983 (

In [40]:
from data.datamapping import *
cols = ['BK', 'BJ', 'VJ', 'HK']
r[[f'B{m}' for m in cols]+[f'J{m}' for m in cols]+[f'K{m}' for m in cols]].style

In [3]:
#mag = ['B', 'V', 'I','J','H','K']
mag = ['V', 'I','J','K']

r = reg['3']
c = '0'
for m in mag:#
    sl = r[f'{m}{c}'].iloc[0]
    serr = r[f'{m}{c}'].iloc[2]
    intr = r[f'{m}{c}'].iloc[1]
    ierr = r[f'{m}{c}'].iloc[3]
    sig = r[f'{m}{c}'].iloc[4]
    #print(f'{m} & {sl:.2f} ({serr:.2f}) & {intr:.2f} ({ierr:.2f})\\\\')
    print(f'\[W_{m}^ (\sigma :{sig: .3f}) = ({sl:.3f} \pm {serr:.3f}) \, (\log P -1) \; + ({intr:.3f} \pm {ierr:.3f}) \] ')
#M_V^ {\mathrm{IRSB}} (\sigma : 0.036) = (-2.281  \pm 0.016) \, ( \log P - 1 ) \; (-3.911 \pm 0.004) 
#M_B^ {\mathrm{IRSB}} (\sigma : 0.046) = (-1.885 \pm 0.020) (\log P -1) + -3.102 \pm 0.005 ]\ 

\[W_V^ (\sigma : 0.202) = (-2.707 \pm 0.101) \, (\log P -1) \; + (14.296 \pm 0.041) \] 
\[W_I^ (\sigma : 0.150) = (-2.960 \pm 0.076) \, (\log P -1) \; + (13.605 \pm 0.030) \] 
\[W_J^ (\sigma : 0.111) = (-3.124 \pm 0.056) \, (\log P -1) \; + (13.216 \pm 0.022) \] 
\[W_K^ (\sigma : 0.086) = (-3.226 \pm 0.043) \, (\log P -1) \; + (12.783 \pm 0.017) \] 


In [4]:
r

Unnamed: 0         V0         I0         J0         K0        VVI  \
0        m_l  -2.706932  -2.960326  -3.124067  -3.225706  -3.430874   
1        c_l  14.295834  13.604655  13.215506  12.783116  12.321155   
2    err_m_l   0.101469   0.075529   0.055883   0.043304   0.085323   
3    err_c_l   0.040531   0.030169   0.022322   0.017297   0.034082   
4     stdd_l   0.202000   0.150000   0.111000   0.086000   0.169000   

         IVI        JVI        KVI        VVJ  ...        JIJ        KIJ  \
0  -3.430874  -3.341223  -3.334419  -3.303033  ...  -3.264391  -3.295958   
1  12.321155  12.623185  12.486543  12.751930  ...  12.882025  12.616114   
2   0.085323   0.048072   0.040524   0.043400  ...   0.049794   0.040496   
3   0.034082   0.019202   0.016187   0.017336  ...   0.019890   0.016176   
4   0.169000   0.095000   0.080000   0.086000  ...   0.099000   0.080000   

         VIK        IIK        JIK        KIK        VJK        IJK  \
0  -3.237692  -3.305320  -3.283295  -3.305320  -3.384557  -3.400722   
1  12.652756  12.536654  12.722583  12.536654  11.413070  11.731128   
2   0.062075   0.038487   0.043801   0.038487   0.044366   0.047829   
3   0.024796   0.015373   0.017496   0.015373   0.017722   0.019105   
4   0.123000   0.076000   0.087000   0.076000   0.088000   0.095000   

         JJK        KJK  
0  -3.327344  -3.327344  
1  12.350726  12.350726  
2   0.035745   0.035745  
3   0.014278   0.014278  
4   0.071000   0.071000  

[5 rows x 29 columns]

In [12]:
reg[[f'{m}0{m}{col}' for m in mag]]
#for i, m in enumerate(mag):
#    print(f"\t{m} &  {reg[m+col].iloc[0] : .3f} ($\pm${reg[m+col].iloc[2] : .3f}) & {reg[m+col].iloc[1] : .3f} ($\pm${reg[m+col].iloc[3] : .3f}) \\\\")

B0BHK         V0VHK         I0IHK         J0JHK         H0HHK  \
0  4.012993e-01  5.104453e-01  7.192397e-01  9.137461e-01  9.655627e-01   
1 -1.950313e-16 -5.638454e-16 -2.271485e-16 -5.805283e-16  2.815759e-16   
2  6.454443e-02  6.559318e-02  5.684106e-02  3.503537e-02  2.316720e-02   
3  3.186029e-02  2.732393e-02  1.945144e-02  1.055896e-02  6.744882e-03   

          K0KHK  
0  9.773706e-01  
1 -8.019632e-16  
2  1.523004e-02  
3  4.434356e-03

In [8]:
import pandas as pd
from data.datamapping import *
mag=['V', 'I', 'J', 'K']
files = {}
files['0'] = '59_madore_3.23'
files['1'] = '76_IRSB_IJ_HK_3.23'#'76_IRSB_IJ_HK_3.23'#'94_jesper_3.23'
files['2'] = '20_cluster_cruz_3.23'
files['3'] = '29_LMC_l_3.4'
files['4'] = '32_SMC_VIJK_s_2.53'
files['5'] = '103_raw_data_IRSB_3.23'
files['6'] = '109_IRSB_plx_JK_p_3.23'

#reg={}
#for k in range(6):
#    reg[str(k)] = pd.read_csv(f'./data/processed/{files[str(k)]}/2_PLPW/{files[str(k)][:2]}_5_regression.csv')

from lvtlaw.h_loadoutput import calibrated_result_
from data.datamapping import *
import pandas as pd
result, reg, res, pre = calibrated_result_()

#print(files[str(k)])
print(file_name)
c='VJ'
for i, m in enumerate(mag):
    sl = reg[f'{m}0{c}S'].iloc[0]
    serr = reg[f'{m}0{c}S'].iloc[2]
    intr = reg[f'{m}0{c}S'].iloc[1]
    ierr = reg[f'{m}0{c}S'].iloc[3]
    sig = reg[f'{m}0{c}S'].iloc[4]

#    print(f'\[M_{m}^{c} (\sigma :{sig: .3f}) = ({sl:.3f} \pm {serr:.3f}) \, (\log P -1) \; + ({intr:.3f} \pm {ierr:.3f}) \] ')
   # print(f'\[M_{m}^{c} = {sl:.3f} \, (\log P -1) \; {intr:.3f} \] ')
   # print(f'\[({sig: .3f})_\sigma (\pm {serr:.3f})_\\alpha \, (\pm {ierr:.3f})_\\beta \] ')
    #for c in ['VK']:
#    print(m)
#    for m in mag:
#        print(c,reg[f'{m}{c}'].iloc[4])
    #print(f"\t{m},{col} &  {reg[m+col].iloc[0] : .3f} ($\pm${reg[m+col].iloc[2] : .3f}) & {reg[m+col].iloc[1] : .3f} ($\pm${reg[m+col].iloc[3] : .3f}) \\\\")

29_LMC_IK
\[M_V^VJ = -2.703 \, (\log P -1) \; 14.294 \] 
\[( 0.061)_\sigma (\pm 0.031)_\alpha \, (\pm 0.012)_\beta \] 
\[M_I^VJ = -2.956 \, (\log P -1) \; 13.603 \] 
\[( 0.014)_\sigma (\pm 0.007)_\alpha \, (\pm 0.003)_\beta \] 
\[M_J^VJ = -3.120 \, (\log P -1) \; 13.214 \] 
\[( 0.070)_\sigma (\pm 0.035)_\alpha \, (\pm 0.014)_\beta \] 
\[M_K^VJ = -3.221 \, (\log P -1) \; 12.781 \] 
\[( 0.070)_\sigma (\pm 0.035)_\alpha \, (\pm 0.014)_\beta \] 


In [1]:
from lvtlaw.h_loadoutput import calibrated_result_
from data.datamapping import *
import pandas as pd
result, reg, res, pre = calibrated_result_()

def table(reg,c):
    print(file_name, c)
    for i, m in enumerate(mag):
        sl = reg[f'{m}0{c}S'].iloc[0]
        serr = reg[f'{m}0{c}S'].iloc[2]
        intr = reg[f'{m}0{c}S'].iloc[1]
        ierr = reg[f'{m}0{c}S'].iloc[3]
        sig = reg[f'{m}0{c}S'].iloc[4]
        print(f'{m} & $({sig:.3f})_\sigma$ &{sl:.3f} & ($\pm$ {serr:.3f}) &{intr:.3f} &($\pm$ {ierr:.3f})\\\\')
#B &  -1.855 ($\pm$ 0.110) & -3.221 ($\pm$ 0.033) \\

table(reg,c='IK')


* * datamapping module loaded!
* * a_utils module loaded!
* * datamapping module loaded!
71_IRSB_plx_IK IK
B & $(0.065)_\sigma$ &-1.966 & ($\pm$ 0.029) &-3.093 &($\pm$ 0.008)\\
V & $(0.029)_\sigma$ &-2.358 & ($\pm$ 0.013) &-3.901 &($\pm$ 0.003)\\
I & $(0.051)_\sigma$ &-2.660 & ($\pm$ 0.023) &-4.682 &($\pm$ 0.006)\\
J & $(0.046)_\sigma$ &-2.884 & ($\pm$ 0.021) &-5.165 &($\pm$ 0.006)\\
H & $(0.055)_\sigma$ &-3.013 & ($\pm$ 0.025) &-5.540 &($\pm$ 0.007)\\
K & $(0.050)_\sigma$ &-3.065 & ($\pm$ 0.023) &-5.600 &($\pm$ 0.006)\\


In [43]:
for i, m in enumerate(mag):
    print(f"\t{m} &  {b[1][m+'0'].iloc[0] : .3f} ($\pm${b[1][m+'0'].iloc[2] : .3f}) & {b[1][m+'0'].iloc[1] : .3f} ($\pm${b[1][m+'0'].iloc[3] : .3f}) \\\\")

	B &  -1.855 ($\pm$ 0.110) & -3.221 ($\pm$ 0.033) \\
	V &  -2.259 ($\pm$ 0.098) & -3.951 ($\pm$ 0.029) \\
	I &  -2.563 ($\pm$ 0.093) & -4.735 ($\pm$ 0.028) \\
	J &  -2.784 ($\pm$ 0.088) & -5.218 ($\pm$ 0.026) \\
	H &  -2.917 ($\pm$ 0.084) & -5.598 ($\pm$ 0.025) \\
	K &  -2.971 ($\pm$ 0.084) & -5.653 ($\pm$ 0.025) \\


In [9]:
from data.datamapping import mag, col_lin
ab=0
col='VI'
f='S'
b[0][[f'{m}{ab}{col}{f}' for m in mag]].style

In [15]:
riess = pd.read_csv('data/18_riess/18_true_abs_data.csv')
riess

Unnamed: 0   name      logP    EBV        plx    M_B0_g    M_V0_g  \
0            0  CFCas  0.687994  0.553  12.252555 -2.254463 -2.910745   
1            1  CSVel  0.771201  0.737  12.547220 -2.616688 -3.218730   
2            2  CVMon  0.730685  0.722  10.978801 -2.437799 -3.005861   
3            3  EVSct  0.490098  0.655  11.225259 -2.706761 -3.196909   
4            4  GHLup  0.967448  0.347  10.222618 -2.846879 -3.706428   
5            5  QZNor  0.578244  0.263  11.425521 -2.782353 -3.406011   
6            6  RSOri  0.878917  0.352  11.001908 -3.115326 -3.724868   
7            7   SNor  0.989194  0.179   9.728963 -3.111366 -3.878133   
8            8  STTau  0.605768  0.368  10.141119 -2.632237 -3.110759   
9            9  SVVul  1.652569  0.462  11.815498 -5.070358 -6.094758   
10          10  TWNor  1.032868  1.157  11.971017 -3.236631 -4.007127   
11          11   USgr  0.828997  0.402   8.923041 -2.838023 -3.531501   
12          12   VCen  0.739882  0.294   9.197433 -2.736436 -3.313053   
13          13   VLac  0.697493  0.335  11.372475 -2.973960 -3.533525   
14          14  AAGem  1.053181  0.309  12.300635 -2.799106 -3.569705   
15          15   XLac  0.735997  0.336  11.290871 -3.406587 -3.971151   
16          16   XPup  1.414321  0.421  11.818071 -3.897448 -4.670901   
17          17   YSct  1.014575  0.757  11.143930 -3.153024 -3.957040   

      M_I0_g    M_J0_g    M_H0_g    M_K0_g  
0  -3.647558 -4.073122 -4.443855 -4.499111  
1  -3.926570 -4.402329 -4.744093 -4.812501  
2  -3.780693 -4.255762 -4.609903 -4.680316  
3  -3.888575 -4.179029 -4.437192 -4.449022  
4  -4.540071 -5.057895 -5.465485 -5.542995  
5  -4.077011 -4.537572 -4.846279 -4.907611  
6  -4.440180 -4.844901 -5.189698 -5.220207  
7  -4.653490 -5.168789 -5.558612 -5.635765  
8  -3.725812 -4.136202 -4.458263 -4.494567  
9  -7.052792 -7.583238 -7.995597 -8.063077  
10 -4.956180 -5.556253 -5.953434 -6.025733  
11 -4.286504 -4.715191 -5.065062 -5.123557  
12 -3.978802 -4.401722 -4.741315 -4.801438  
13 -4.164361 -4.567434 -4.897326 -4.914239  
14 -4.326461 -4.858071 -5.281285 -5.313405  
15 -4.633721 -4.999772 -5.330306 -5.374019  
16 -5.537848 -6.035141 -6.464200 -6.548891  
17 -4.789557 -5.279902 -5.690495 -5.743898

In [2]:
# madore result
Band & Slope  & $\sigma $ (mag)  & Zero  & $\sigma$(mag)  & rms \\
B & −2.277 & 0.121 & −3.214 & 0.036 & 0.27 //
V& −2.670 & 0.095 & −3.944  & 0.028  & 0.21 //
R& −2.874 & 0.075 & −4.396  & 0.022 &  0.17 //
I& −2.983 & 0.064 & −4.706  & 0.019  & 0.14 //
J& −3.198 &  0.054 & −5.258 &  0.016 &  0.12 //
H& −3.333 & 0.055 & −5.558  & 0.017  & 0.12 //
K& −3.377 & 0.055 & −5.659  & 0.016 &  0.12 //
W, VI & −3.476 & 0.037 & −5.889 &  0.011  & 0.08 //
W, BI & −3.600 & 0.033 & −5.997 &  0.010  & 0.07  //

$$
\begin{array}{|cc|}
\hline
M_B & = -1.86(\log P - 1)(\pm 0.011)  -3.22(\pm 0.003)\\
M_V & = -2.26(\log P - 1)(\pm 0.003)  -3.95(\pm 0.001)\\
M_I & = -2.57(\log P - 1)(\pm 0.014)  -4.74(\pm 0.004)\\
M_J & = -2.79(\log P - 1)(\pm 0.011)  -5.22(\pm 0.003)\\
M_H & = -2.92(\log P - 1)(\pm 0.011)  -5.60(\pm 0.003)\\
M_K & = -2.97(\log P - 1)(\pm 0.011)  -5.65(\pm 0.003)\\
\hline
M_B & = -1.85(\log P - 1)(\pm 0.009)  -3.22(\pm 0.003)\\
M_V & = -2.26(\log P - 1)(\pm 0.010)  -3.95(\pm 0.003)\\
M_I & = -2.56(\log P - 1)(\pm 0.018)  -4.73(\pm 0.005)\\
M_J & = -2.78(\log P - 1)(\pm 0.010)  -5.22(\pm 0.003)\\
M_H & = -2.92(\log P - 1)(\pm 0.014)  -5.60(\pm 0.004)\\
M_K & = -2.97(\log P - 1)(\pm 0.014)  -5.65(\pm 0.004)\\
\hline
\end{array}
$$



$$
\begin{array}{|cc|}
\hline
M_B & = -1.86(\log P - 1)(\pm 0.011)  -3.22(\pm 0.003)\\
M_V & = -2.26(\log P - 1)(\pm 0.003)  -3.95(\pm 0.001)\\
M_I & = -2.57(\log P - 1)(\pm 0.014)  -4.74(\pm 0.004)\\
M_J & = -2.79(\log P - 1)(\pm 0.011)  -5.22(\pm 0.003)\\
M_H & = -2.92(\log P - 1)(\pm 0.011)  -5.60(\pm 0.003)\\
M_K & = -2.97(\log P - 1)(\pm 0.011)  -5.65(\pm 0.003)\\
\hline
M_B & = -1.85(\log P - 1)(\pm 0.009)  -3.22(\pm 0.003)\\
M_V & = -2.26(\log P - 1)(\pm 0.010)  -3.95(\pm 0.003)\\
M_I & = -2.56(\log P - 1)(\pm 0.018)  -4.73(\pm 0.005)\\
M_J & = -2.78(\log P - 1)(\pm 0.010)  -5.22(\pm 0.003)\\
M_H & = -2.92(\log P - 1)(\pm 0.014)  -5.60(\pm 0.004)\\
M_K & = -2.97(\log P - 1)(\pm 0.014)  -5.65(\pm 0.004)\\
\hline
\end{array}
$$
